tests/Doctrine/Tests/DBAL/Schema/IndexTest.php

```diff
@@ -188,4 +189,40 @@ public function testOptions() : void
         self::assertSame('name IS NULL', $idx2->getOption('WHERE'));
         self::assertSame(['where' => 'name IS NULL'], $idx2->getOptions());
     }
+
+    /**
+     * @param int[] $lengths1
+     * @param int[] $lengths2
+     *
+     * @dataProvider spatialIndexProvider
+     */
+    public function testFulfilledSpatialIndex(?string $flag1, ?string $flag2, array $lengths1, array $lengths2, bool $expected) : void
+    {
+        $index1 = new Index('index1', ['geo_location'], false, false, [], ['lengths' => $lengths1]);
+        if ($flag1 !== null) {
```

Comment:
Please avoid flow control in tests. Would it be possible to pass flags as an empty or non-empty array?

Reply:
Good idea